framework/audio/internal/mixer.h

```diff
@@ -0,0 +1,93 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_MIXER_H
+#define MU_AUDIO_MIXER_H
+
+#include <memory>
+#include <map>
+#include "iaudiosource.h"
+#include "internal/mixerchannel.h"
+#include "internal/clock.h"
+#include "rpc/itarget.h"
+
+namespace mu {
+namespace audio {
+class Mixer : public IAudioSource, public rpc::ITarget
+{
+    enum Mode {
+        MONO,
+        STEREO,
+//TODO: SURROUND, //NOTE: 5.1, 7.1, etc
+//TODO: SPATIAL //NOTE: 2 channel output with HRTF
+    };
+
+public:
+    Mixer();
+    ~Mixer();
+
+    Mode mode() const;
+    void setMode(const Mode& mode);
+
+    void setClock(std::shared_ptr<Clock> clock);
+
+    virtual void setSampleRate(unsigned int sampleRate) override;
+
+    unsigned int streamCount() const override;
+
+    //!set master level
+    void setLevel(float level);
+
+    //! return insert at master
+    std::shared_ptr<IAudioInsert> insert(unsigned int number) const;
+
+    //! set master insert
+    void setInsert(unsigned int number, std::shared_ptr<IAudioInsert> insert);
+
+    //! add source to the mix
+    unsigned int addChannel(std::shared_ptr<IAudioSource> source);
```

Comment:
maybe addSource?

Reply:
And why does this method returns int?

Reply:
@Eism no. this method adds new channel to the mixer and connect it to the given soure
@RomanPudashkin it returns channel's number